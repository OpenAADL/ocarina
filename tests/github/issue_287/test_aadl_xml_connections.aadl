package Test_Aadl_Xml_Connections
public

    data dummy_data end dummy_data;

    system main
    end main;

    system implementation main.impl
        subcomponents
            p1: process dummy_process.impl;
            p2: process dummy_process.impl;

        connections
            conn1: port p1.out_port -> p2.in_port {Timing => Immediate; Latency => 1ms .. 10ms;};
    end main.impl;

    process dummy_process
        features
            in_port: in data port dummy_data;
            out_port: out data port dummy_data;
    end dummy_process;

    process implementation dummy_process.impl
        subcomponents
            t1: thread dummy_thread.impl;
            t2: thread dummy_thread.impl;

        connections
            conn1: port in_port -> t1.in_port;
            conn2: port t1.out_port -> t2.in_port {Timing => Delayed;};
            conn3: port t2.out_port -> out_port;
    end dummy_process.impl;

    thread dummy_thread
        features
            in_port: in data port dummy_data;
            out_port: out data port dummy_data;
    end dummy_thread;

    thread implementation dummy_thread.impl
    end dummy_thread.impl;

end Test_Aadl_Xml_Connections;