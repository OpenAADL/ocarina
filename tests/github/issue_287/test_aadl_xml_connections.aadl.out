<aadl_xml root_system="main.impl">
 <components>
   <component category="system" identifier="main.impl">
     <classifier>
       main.impl     </classifier>
     <features/>
     <subcomponents>
       <component category="bus" identifier="b1">
         <classifier>
           dummy_bus.impl         </classifier>
         <features/>
         <subcomponents/>
         <properties/>
       </component>
       <component category="process" identifier="p1">
         <classifier>
           dummy_process.impl         </classifier>
         <features>
           <feature identifier="in_port">
             <direction kind="in"/>
             <type kind="data"/>
             <corresponding_instance>
               <component category="data" identifier="dummy_data">
                 <classifier>
                   dummy_data                 </classifier>
                 <features/>
                 <subcomponents/>
                 <properties/>
               </component>
             </corresponding_instance>
           </feature>
           <feature identifier="out_port">
             <direction kind="out"/>
             <type kind="data"/>
             <corresponding_instance>
               <component category="data" identifier="dummy_data">
                 <classifier>
                   dummy_data                 </classifier>
                 <features/>
                 <subcomponents/>
                 <properties/>
               </component>
             </corresponding_instance>
           </feature>
         </features>
         <subcomponents>
           <component category="thread" identifier="t1">
             <classifier>
               dummy_thread.impl             </classifier>
             <features>
               <feature identifier="in_port">
                 <direction kind="in"/>
                 <type kind="data"/>
                 <corresponding_instance>
                   <component category="data" identifier="dummy_data">
                     <classifier>
                       dummy_data                     </classifier>
                     <features/>
                     <subcomponents/>
                     <properties/>
                   </component>
                 </corresponding_instance>
               </feature>
               <feature identifier="out_port">
                 <direction kind="out"/>
                 <type kind="data"/>
                 <corresponding_instance>
                   <component category="data" identifier="dummy_data">
                     <classifier>
                       dummy_data                     </classifier>
                     <features/>
                     <subcomponents/>
                     <properties/>
                   </component>
                 </corresponding_instance>
               </feature>
             </features>
             <subcomponents/>
             <properties/>
           </component>
           <component category="thread" identifier="t2">
             <classifier>
               dummy_thread.impl             </classifier>
             <features>
               <feature identifier="in_port">
                 <direction kind="in"/>
                 <type kind="data"/>
                 <corresponding_instance>
                   <component category="data" identifier="dummy_data">
                     <classifier>
                       dummy_data                     </classifier>
                     <features/>
                     <subcomponents/>
                     <properties/>
                   </component>
                 </corresponding_instance>
               </feature>
               <feature identifier="out_port">
                 <direction kind="out"/>
                 <type kind="data"/>
                 <corresponding_instance>
                   <component category="data" identifier="dummy_data">
                     <classifier>
                       dummy_data                     </classifier>
                     <features/>
                     <subcomponents/>
                     <properties/>
                   </component>
                 </corresponding_instance>
               </feature>
             </features>
             <subcomponents/>
             <properties/>
           </component>
         </subcomponents>
         <properties/>
         <connections>
           <connection name="conn1" type="port_connection">
             <src feature="in_port"/>
             <dst component="t1" feature="in_port"/>
             <properties/>
           </connection>
           <connection name="conn2" type="port_connection">
             <src component="t1" feature="out_port"/>
             <dst component="t2" feature="in_port"/>
             <properties>
               <property name="Timing">
                 <property_value>
                   <enumeration value="Delayed"/>
                 </property_value>
               </property>
             </properties>
           </connection>
           <connection name="conn3" type="port_connection">
             <src component="t2" feature="out_port"/>
             <dst feature="out_port"/>
             <properties/>
           </connection>
         </connections>
       </component>
       <component category="process" identifier="p2">
         <classifier>
           dummy_process.impl         </classifier>
         <features>
           <feature identifier="in_port">
             <direction kind="in"/>
             <type kind="data"/>
             <corresponding_instance>
               <component category="data" identifier="dummy_data">
                 <classifier>
                   dummy_data                 </classifier>
                 <features/>
                 <subcomponents/>
                 <properties/>
               </component>
             </corresponding_instance>
           </feature>
           <feature identifier="out_port">
             <direction kind="out"/>
             <type kind="data"/>
             <corresponding_instance>
               <component category="data" identifier="dummy_data">
                 <classifier>
                   dummy_data                 </classifier>
                 <features/>
                 <subcomponents/>
                 <properties/>
               </component>
             </corresponding_instance>
           </feature>
         </features>
         <subcomponents>
           <component category="thread" identifier="t1">
             <classifier>
               dummy_thread.impl             </classifier>
             <features>
               <feature identifier="in_port">
                 <direction kind="in"/>
                 <type kind="data"/>
                 <corresponding_instance>
                   <component category="data" identifier="dummy_data">
                     <classifier>
                       dummy_data                     </classifier>
                     <features/>
                     <subcomponents/>
                     <properties/>
                   </component>
                 </corresponding_instance>
               </feature>
               <feature identifier="out_port">
                 <direction kind="out"/>
                 <type kind="data"/>
                 <corresponding_instance>
                   <component category="data" identifier="dummy_data">
                     <classifier>
                       dummy_data                     </classifier>
                     <features/>
                     <subcomponents/>
                     <properties/>
                   </component>
                 </corresponding_instance>
               </feature>
             </features>
             <subcomponents/>
             <properties/>
           </component>
           <component category="thread" identifier="t2">
             <classifier>
               dummy_thread.impl             </classifier>
             <features>
               <feature identifier="in_port">
                 <direction kind="in"/>
                 <type kind="data"/>
                 <corresponding_instance>
                   <component category="data" identifier="dummy_data">
                     <classifier>
                       dummy_data                     </classifier>
                     <features/>
                     <subcomponents/>
                     <properties/>
                   </component>
                 </corresponding_instance>
               </feature>
               <feature identifier="out_port">
                 <direction kind="out"/>
                 <type kind="data"/>
                 <corresponding_instance>
                   <component category="data" identifier="dummy_data">
                     <classifier>
                       dummy_data                     </classifier>
                     <features/>
                     <subcomponents/>
                     <properties/>
                   </component>
                 </corresponding_instance>
               </feature>
             </features>
             <subcomponents/>
             <properties/>
           </component>
         </subcomponents>
         <properties/>
         <connections>
           <connection name="conn1" type="port_connection">
             <src feature="in_port"/>
             <dst component="t1" feature="in_port"/>
             <properties/>
           </connection>
           <connection name="conn2" type="port_connection">
             <src component="t1" feature="out_port"/>
             <dst component="t2" feature="in_port"/>
             <properties>
               <property name="Timing">
                 <property_value>
                   <enumeration value="Delayed"/>
                 </property_value>
               </property>
             </properties>
           </connection>
           <connection name="conn3" type="port_connection">
             <src component="t2" feature="out_port"/>
             <dst feature="out_port"/>
             <properties/>
           </connection>
         </connections>
       </component>
       <component category="device" identifier="d1">
         <classifier>
           dummy_device.impl         </classifier>
         <features>
           <feature identifier="b1">
             <direction kind="none"/>
             <type kind="access"/>
             <corresponding_instance>
               <component category="bus" identifier="dummy_bus">
                 <classifier>
                   dummy_bus                 </classifier>
                 <features/>
                 <subcomponents/>
                 <properties/>
               </component>
             </corresponding_instance>
           </feature>
         </features>
         <subcomponents/>
         <properties/>
       </component>
     </subcomponents>
     <properties/>
     <connections>
       <connection name="conn1" type="port_connection">
         <src component="p1" feature="out_port"/>
         <dst component="p2" feature="in_port"/>
         <properties>
           <property name="Latency">
             <property_value>
               <integer_range_unit value_low="1" value_high="10" unit_low="ms" unit_high="ms"/>
             </property_value>
           </property>
           <property name="Timing">
             <property_value>
               <enumeration value="Immediate"/>
             </property_value>
           </property>
         </properties>
       </connection>
       <connection name="conn2" type="access_bus">
         <src feature="b1"/>
         <dst component="d1" feature="b1"/>
         <properties/>
       </connection>
     </connections>
   </component>
 </components>
</aadl_xml>
